gobblin-modules/gobblin-kafka-common/src/main/java/org/apache/gobblin/metrics/reporter/KeyValueEventObjectReporter.java

```diff
@@ -83,17 +81,24 @@ public KeyValueEventObjectReporter(Builder builder) {
           "Key not assigned from config. Please set it with property {} Using randomly generated number {} as key ",
           ConfigurationKeys.METRICS_REPORTING_EVENTS_PUSHERKEYS, randomKey);
     }
+
+    schema = AvroUtils.overrideNameAndNamespace(GobblinTrackingEvent.getClassSchema(), builder.topic, builder.namespaceOverride);
   }
 
   @Override
   public void reportEventQueue(Queue<GobblinTrackingEvent> queue) {
-    log.info("Emitting report using KeyValueEventObjectReporter");
 
     List<Pair<String, GenericRecord>> events = Lists.newArrayList();
     GobblinTrackingEvent event;
 
     while (null != (event = queue.poll())) {
-      GenericRecord record = AvroUtils.overrideNameAndNamespace(event, this.topic, this.namespaceOverride);
+
+      GenericRecord record=event;
+      try {
+        record = AvroUtils.convertRecordSchema(event, schema);
```

Comment:
This conversion is expensive to do on every event. This is the same behavior as today, so it is not making things worse, but we should consider supporting pluggable compiled schemas to avoid the need to override the namespace.

Reply:
Its not on every event but every new instance of the reporter. So it's probably the order of mappers than number of messages sent out which is probably in the order of thousands.